### ISSUE TITLE
Fenix New Tab Experiment (Test branch 3: Default Browser Message to a new user on their third session, Control: No Message)

### ISSUE THREAD

**Issue by ItsDevyani** [mozilla-mobile/fenix#18375]
## User Story
- As a user, I want to know how I can easily set up Fenix as my default browser once I've had a chance to use the browser a few times.

## Dependencies
- Test Measurement plan is being requested by the DS team
- Default Browser Dashboard is also being requested by the DS team
- Nimbus SDK integration with Fenix 
-[ UX design work is done](https://www.figma.com/file/YvvFxwXKonLkvdF0ZcL63Y/Design-feedback?node-id=0%3A1)
- Engineering Time

### Acceptance Criteria
- A Message/Notification can be triggered to a user on their third session with Fenix.
- A user can click the notification and be launched into the Default Browser flow.
- Funnel Activity (Clicks on each step) and Conversion to the default browser feature can be measured.
- Nimbus SDK integration with New Tab allowing experimentation on it

### Engineering notes
- [Nimbus engineering docs](https://mozilla.github.io/experimenter-docs/fenix-engineers/)
- Please sync with data science before starting this implementation to confirm the following:

**Experiment length**: TBD (default: 35 days total, 7 day enrollment period)
**Start date**: TBD
**Experiment size**: TBD
**A/B split**: TBD (default: 50/50)
**Channels**: Release, Nightly
   * > all of the Nimbus experiments should be pushed to nightly first. [...]  Also, they don't need to go to beta afterwards but instead can go to production.

**Comment by ItsDevyani** [comment]
Project Update: 
- Nicole Weber has made the designs for this experiment and they are ready to be shared as needed.
- We have has discussions with the Nimbus Team to launch this as a Nimbus Experiment.

**Comment by amedyne** [comment]
Note: This ticket will be for tracking the Engineering tasks for developing Experiment **Test Branch 3: Default Browser Message to a new user on their third session** and the corresponding telemetry needed for this branch.

**Comment by amedyne** [comment]
Tagging @topotropic , too.

**Comment by ItsDevyani** [comment]
Is this influencing the experiment?

_We are comparing three tactics against each other, this is one of those 3 tactics, so showing a banner on the new tab to a new user on their 3rd app open after installing the app is a factor that influences the experiment._

Another question: if we keep the timing, should the message be shown on the home screen only on the third session, or in each one from that point forward? In the Figma designs, we have a close button. Is clicking this supposed to dismiss the message and also prevent it from showing again? (again, this differs from the "fixed" options in the other branches).

_Once the message shows, it should keep showing till the user either clicks the 'x' button to dismiss or clicks the 'set to default' button. once the message clears it gone forever._

I am ok with either decision, I would just like to note that if we would show the message only on the third session there might be cases where a user never sees the message, as opening the browser on the last visited tab will skip showing the home screen.

_noted that makes sense._

**Comment by mcarare** [comment]
Another question: Is this supposed to be shown in private browsing too?

**Comment by amedyne** [comment]
> Another question: Is this supposed to be shown in private browsing too?

Per @ItsDevyani , no notification in private browsing.

**Comment by lobontiumira** [comment]
Hi all,

I was able to trigger the "Set as default browser" card on the homescreen, after the following steps:
- open the app, browse a page at least, close the app, not just background it,
- open the app for the second time, do the same actions as in the previous step,
- open the third session, browse a few pages, close again the app, not just background it,
- the fourth time, when opened, go to the homescreen, and the "Set as default browser" card is displayed.

Tapping on the "Set as default browser" button, redirect the user to set the default browser.
Tapping on the "x" from the card, dismisses the card, and does not show it again.

Tested on Google Pixel (Android 10), and OnePlus 5T (Android 10), on a debug Nightly build #2015805929, GV 89.0a1-20210418092832.

Note: even if clearing the data from the device's settings-apps, the card on the homescreen is no longer displayed. Uninstalling and reinstalling the app, redisplayes the card.


https://user-images.githubusercontent.com/40694158/115868609-60e12a80-a445-11eb-9ab9-a590e90ce306.mp4

@ItsDevyani should I test anything else regarding this issue?
Thank you!
